src/mg_solver/HpMultiGrid.cpp

```diff
@@ -737,13 +738,12 @@ MultiGrid::bottomsolve ()
 void
 MultiGrid::average_down_acoef (FArrayBox const& a_acf)
 {
-#if defined(AMREX_USE_GPU)
-    Gpu::dtod_memcpy_async(m_acf[0].dataPtr(), a_acf.dataPtr(), m_acf[0].nBytes());
-#else
-    Real const* psrc = a_acf.dataPtr();
-    Real * pdst = m_acf[0].dataPtr();
-    hpmg::ParallelFor(m_acf[0].size(), [=] (Long i) noexcept { pdst[i] = psrc[i]; });
-#endif
+    auto const& array_m_acf = m_acf[0].array();
+    auto const& array_a_acf = a_acf.const_array();
```

Comment:
Why is the memcpy no longer needed?

Reply:
```
#if defined(AMREX_USE_GPU)
    Gpu::dtod_memcpy_async(m_acf[0].dataPtr(), a_acf.dataPtr(), m_acf[0].nBytes());
#else
    Real const* psrc = a_acf.dataPtr();
    Real * pdst = m_acf[0].dataPtr();
    hpmg::ParallelFor(m_acf[0].size(), [=] (Long i) noexcept { pdst[i] = psrc[i]; });
#endif
```
was replaced with
```
    auto const& array_m_acf = m_acf[0].array();
    auto const& array_a_acf = a_acf.const_array();
    hpmg::ParallelFor(m_acf[0].box(), [=] AMREX_GPU_DEVICE (int i, int j, int) noexcept
        {
            array_m_acf(i,j,0) = array_a_acf(i,j,0);
        });
```
as now even for GPUs memcpy can’t be used anymore because the a_acf array has ghost cells and m_acf does not.